File: LeetCode/Python/Medium/881_boats_to_save_people.py
class Solution:
    def numRescueBoats(self, people: List[int], limit: int) -> int:
        people = sorted(people)
        i, j = 0, len(people) - 1
        num_boats = 0
        while i <= j:
            if i == j:
                i += 1
            elif people[i] + people[j] <= limit:
                i += 1
                j -= 1
            elif people[i] + people[j] > limit:
                j -= 1
            num_boats += 1
        return num_boats

File: LeetCode/Javascript/Medium/881_boats_to_save_people.js
const numRescueBoats = (people, limit) => {
    let i = 0;
    let j = people.length - 1;
    let numBoats = 0;
    people.sort((a, b) => a - b);
    while(i <= j){
        if(i === j){
            i++;
        } else if(people[i] + people[j] <= limit){
            i++;
            j--;
        } else if(people[i] + people[j] > limit){
            j--;
        }
        numBoats++;
    }
    return numBoats;
};